File: Temporada02/Episodio01/Ruby/Ejemplo3.rb
def same_var_assigned_in_closure

  temp1 = 1
  a_closure = lambda { | ;temp1|
    temp1 = 10
    temp1 + 2 }
  [a_closure.call, temp1]

end

# Modifica temp1 o crea una nueva var temp1?
print same_var_assigned_in_closure
# Se puede definir una temp1 local al closure?

File: Temporada02/Episodio01/JavaScript/Ejemplo3.js
function same_var_assigned_in_closure() {

    let temp1 = 1
    const a_closure = () => {
        let temp1 = 10
        return temp1 + 2
    }

    return [a_closure(), temp1]
}

console.log(same_var_assigned_in_closure())
// Se puede definir temp1 como local al closure?
